### PR TITLE
Task 10. GET /api/articles (queries)

### PR DESCRIPTION
**Pull request by IzaanD98**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -291,20 +291,139 @@ describe("/api/articles/:article_id", () => {
...
+  });
+  it("400: GET - responds 400 status code if sort_by column is invalid", () => {
+    return request(app)
+      .get("/api/articles?topic=banana")
+      .expect(400)
+      .then(({ body }) => {
+        const error = body.message;
+        expect(error).toBe("Invalid column");
+      });
+  });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:340]: This should be a 404 for a valid but non-existent topic

File: __tests__/app.test.js [hunk 2]
```diff
@@ -291,20 +291,139 @@ describe("/api/articles/:article_id", () => {
...
+    return request(app)
+      .get("/api/articles?order=hello")
+      .expect(400)
+      .then(({ body }) => {
+        const error = body.message;
+        expect(error).toBe("Invalid order query");
+      });
+  });
+});
+
```
Comment (johnhutcheon) [inline __tests__/app.test.js:414]: Could we please add one more status 200 test for when a topic query is valid, but has no articles, so returns an empty array?

File: models/newsModel.js [hunk 3]
```diff
@@ -75,7 +75,53 @@ exports.getAllUsers = () => {
   const queryString = `SELECT * FROM users`;
 
   return db.query(queryString).then((results) => {
-    console.log(results.rows);
     return results.rows;
   });
 };
+
+exports.getQueriedArticles = (topic, sort_by, order) => {
```
Comment (johnhutcheon) [inline models/newsModel.js:82]: Could we add some default values for topic and sort_by if the values are undefined please?

**Review by johnhutcheon — APPROVED** [review]
Great work Izaan.
Happy for you to merge, but could you just check my comment and make amendments please?
